test/unit/test_manager.py

```diff
@@ -19,6 +19,24 @@ def test_connect_ssh(self, mock_ssh):
         manager.connect(host='host')
         mock_ssh.assert_called_once_with(host='host')
 
+    @patch('ncclient.manager.connect_ssh')
+    def test_connect_ssh_with_hostkey_ed25519(self, mock_ssh):
+        hostkey = 'AAAAC3NzaC1lZDI1NTE5AAAAIIiHpGSf8fla6tCwLpwshvMGmUK+B/0v5CsRu+5v4uT7'
+        manager.connect(host='host', hostkey=hostkey)
+        mock_ssh.assert_called_once_with(host='host', hostkey=hostkey)
```

Comment:
Testing feels a little thin here. 

- Do we have the ability to check that `_preferred_keys` is set correctly in each case? 
- Test for host key lookup when port is non-default?
- Negative test for SSHUnknownHostError when key doesn't match?

Reply:
Yep more tests would be great. I feel like what's there, is in line with precedents though. Might have a look later but are you OK with this PR without @glennmatthews ?